volume/mounts/windows_parser.go

```diff
@@ -246,8 +247,22 @@ func (p *windowsParser) validateMountConfigReg(mnt *mount.Mount, additionalValid
 		if err != nil {
 			return &errMountConfig{mnt, err}
 		}
+
+		if windowsDetectMountType(mnt.Target) == mount.TypeNamedPipe {
+			return &errMountConfig{mnt, fmt.Errorf("'%s' is not a valid bind path", mnt.Target)}
+		}
+
 		if !exists {
-			return &errMountConfig{mnt, errBindSourceDoesNotExist(mnt.Source)}
+			if isLongFormBind {
+				// We don't allow src direction for long form bind requests.
+				// docker run --rm -it --mount type=bind,src=C:\hello,dst=C:\world mcr.microsoft.com/windows/servercore:ltsc2019 powershell
+				return &errMountConfig{mnt, errBindSourceDoesNotExist(mnt.Source)}
+			}
+			if err := system.MkdirAllWithACL(mnt.Source, 0, system.SddlAdministratorsLocalSystem); err != nil {
```

Comment:
A method named `validate` associated with a `parser` type should not have side effects.